File: Spoj/main/RLM.cpp
#include<iostream>
#include<cstdio>
#include<cstring>
#include<cstdlib>
using namespace std;
long long toDecimal(char *s){
   char dec[200],times,fill,top=0;
   long long ret;
   for(int i=0;s[i]!=0;i+=2){
      times=s[i];
      fill=s[i+1];
      for(int k='0';k<times;k++)dec[top++]=fill;
   }
   dec[top]='\0';
   ret=atoll(dec);
   return ret;
}
void toRlm(long long num){
   char s[200],ret[500],val;
   int streak=1,top=0;
   sprintf(s,"%lld",num);
   val=s[0];
   for(int i=1;s[i]!='\0';i++){
      if(s[i]==val){
         streak++;
         if(streak==10){
            ret[top++]='9';
            ret[top++]=val;
            streak=1;
         }
      }
      else{
         ret[top++]='0'+streak;
         ret[top++]=val;
         streak=1;
         val=s[i];
      }
   }
   ret[top++]='0'+streak;
   ret[top++]=val;
   ret[top]='\0';
   printf("%s\n",ret);
}
int main(){
   char a[200];
   char b[200],opr;
   long long num1,num2;
   while(scanf("%s %c %s",a,&opr,b)>0){
      //cout<<opr;
      num1=toDecimal(a);
      num2=toDecimal(b);
      printf("%s %c %s = ",a,opr,b);
      if(opr=='+')toRlm(num1+num2);
      else if(opr=='-')toRlm(num1-num2);
      else if(opr=='*')toRlm(num1*num2);
      else if(opr=='/')toRlm(num1/num2);
   }   
   return 0;
}

File: Spoj/main/RLM.py
#import sys
#s=sys.stdin.read().split('\n')
def toDecimal(n,l):
    ret=''
    for i in range(0,l,2):
        ret+=(n[i+1]*int(n[i]))
    return int(ret)
def toRLM(n):
    #print n
    prev=n%10
    n//=10
    count=1
    ret=''
    if n==0: return '1'+str(prev)
    while n>10:
        temp=n%10
        n//=10
        if temp==prev:count+=1
        else:
            if count>=9:
                    while count>=9:
                        ret+=(str(prev)+'9')
                        count-=9
                    if count>0:ret=(str(prev)+str(count))+ret
                    prev,count=temp,1
	    else:
                ret+=(str(prev)+str(count))
                prev,count=temp,1
    if n==prev:count+=1
    #print ret,count
    while count>=9:
        ret+=(str(prev)+'9')
        count-=9
    if count>0:ret=(str(prev)+str(count))+ret
    if n !=prev: ret+=(str(n)+'1')
    return ret[::-1]
'''for line in s:
    try:
        inp=line.split()        
        num1=toDecimal(inp[0],len(inp[0]))
        num2=toDecimal(inp[2],len(inp[2]))
        op=inp[1]
        if op=='+':print '%s + %s = %s' % (inp[0],inp[2],toRLM(num1+num2))
        elif op=='-':print '%s - %s = %s' % (inp[0],inp[2],toRLM(num1-num2))
        elif op=='*':print '%s * %s = %s' % (inp[0],inp[2],toRLM(num1*num2))
        else:print '%s / %s = %s' % (inp[0],inp[2],toRLM(num1//num2))
    except:break'''
while 1:
  print toRLM(input())

